### DISCUSSION TITLE
Keep existing Mongo data up-to-date as schema changes?

### DISCUSSION THREAD

**Discussion by elramus** [prisma/prisma#13624] [Q&A] [answered]
I'm wondering what the recommended approach is to keeping a Mongo collection up-to-date as the schema changes over time. Let's say we start with:
```
model User {
  id String @id @default(auto()) @map("_id") @db.ObjectId
  email String
}
```
and then down the road (after many `User`s have already been created), we need to add:
```
model User {
  ...
  wantsNewsletter Boolean
}
```
Unless I'm missing a step somewhere, I add this to my Prisma schema and then when I run `npx prisma db push` it tells me that the database is already in sync with the schema. Is that expected? I'm coming from the SQL world and Mongo is still pretty new to me, but I would expect something akin to what Django does and says that you can't just go adding a new field without providing a default. I could then add `@default(false)` to the schema, and Prisma could add a `wantsNewsletter` field to all my existing `User` records with the default value.

Same thing with if I was to change the `email` field to be called `emailAddress` instead. Django would ask if I'm simply renaming, and then go about changing existing data accordingly. Otherwise it seems like we're sort of defeating the purpose of type safety in the first place, if there's no way to ensure records stay in a particular shape over time. 

What's the Mongo-y or Prisma-y way to ensure that my records don't totally go off the rails whenever there are schema changes after data's already in there? Or is that just what you get when you use NoSQL and one should always just assume every field is potentially `undefined`?

Thanks!

**Comment by austincrim** [comment, marked as answer]
Hey @elramus 👋,

This kind of behavior comes with the territory of using MongoDB 😄. It does not enforce a strict schema at the database level, so it is up to the developer to keep the data clean over time.

If you are using Prisma from the beginning, we will at least enforce required fields when writing data, but you will always need to be mindful when adding new fields to a model.

**Reply by janpio** [reply]
If you add that `@default(false)` Prisma will use that for missing `wantsNewsletter` values on read and fill in `false`. That way your data will get consistent over time - but of course only when using Prisma as the client talking to your MongoDB instance.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
